### ISSUE TITLE
Evaluating on BBH/MMLU using some of the FT checkpoints from the model-zoo. 

### ISSUE THREAD

**Issue by prateeky2806** [Alpha-VLLM/LLaMA2-Accessory#81]
Hi, thank you for creating this amazing repository. I have two questions. 

1. I am trying to use some of the existing models listed in the [model-zoo](https://llama2-accessory.readthedocs.io/en/latest/inference.html#model-zoo) and run evaluations on them. However, I am not able to figure out how this can be done. Can you provide a script that can do this? 
2. I am wondering if you have some QLora Finetuned checkpoints that are also provided by you. I want to load these QLora modules and then perform BBH/MMLU evaluation on the model. 

It would be great if you can help with this. 

Thanks,
Prateek

**Comment by kriskrisliu** [comment]
Hi there,

Thank you for your response. I've reviewed the information you provided, and everything seems to be in order.

1. You can find the evaluation guidance on [this page](https://llama2-accessory.readthedocs.io/en/latest/light-eval/index.html).

2. Any of the checkpoints in the model zoo can be loaded with the quantization method by using [this function](https://github.com/Alpha-VLLM/LLaMA2-Accessory/blob/d66fd91a68c234b4be9955266fa28b5a2a44105c/accessory/util/quant.py#L95). I noticed that the evaluation team was not involved in the quantization implementation. I plan to submit a pull request this weekend to address this.

If you have any further questions or concerns, please feel free to let me know.

**Comment by kriskrisliu** [comment]
I've checked the code and revisited the BBH implementation. Other evaluation codes are in the revisiting progress.
If you would like to evaluate on BBH, the running script could be like the following. In this case, it initializes a 7b llama2 model, and loads [official llama2 weights](https://huggingface.co/meta-llama/Llama-2-7b) as well as PEFT weights ([3.35MB, alpaca_llamaPeft_normBias](https://huggingface.co/Alpha-VLLM/LLaMA2-Accessory/blob/main/finetune/sg/alpaca_llamaPeft_normBias/consolidated.00-of-01.model-diff.pth)). Since the `--quant` flag is activated, the base model will be quantized to 4bit by using QLoRA implementation, while the PEFT weights are stayed in higher precision(fp32/fp16/bp16).
```bash
cd LLaMA2-Accessory/light-eval
torchrun --nproc-per-node=1 --master_port 23456 src/eval_bbh.py \
--llama_type llama_peft \
--llama_config <path-to>/Llama-2-7b/params.json \
--tokenizer_path <path-to>/Llama-2-7b/tokenizer.model \
--pretrained_path <path-to>/Llama-2-7b/   <path-to>/alpaca_llamaPeft_normBias \
--data_dir data/BIG-Bench-Hard \
--quant
```

**Comment by prateeky2806** [comment]
Thank you so much! I was wondering if you could provide a way to find out which checkpoints are Lora/Qlora checkpoints and which are fully finetuned models so that it is easier for people to select the correct models from the model zoo.

**Comment by ChrisLiu6** [comment]
> Thank you so much! I was wondering if you could provide a way to find out which checkpoints are Lora/Qlora checkpoints and which are fully finetuned models so that it is easier for people to select the correct models from the model zoo.

Thank you for your suggestion. In fact, we have been doing so. Parameter efficient fine-tuning checkpoints are generally labeled with "peft" or "llamaAdapter", e.g. [alpacaLlava_llamaQformerv2Peft_13b](https://huggingface.co/Alpha-VLLM/LLaMA2-Accessory/tree/main/finetune/mm/alpacaLlava_llamaQformerv2Peft_13b) and [alpaca_llamaPeft_normBias](https://huggingface.co/Alpha-VLLM/LLaMA2-Accessory/tree/main/finetune/sg/alpaca_llamaPeft_normBias), and those without special labels are full-parameter fine-tune methods
